iina/Logger.swift

```diff
@@ -20,9 +30,7 @@ class Logger {
     case warning
     case error
 
-    static var preferred: LogLevel {
-      return LogLevel(rawValue: Preference.integer(for: .logLevel).clamped(to: 0...3))!
-    }
+    static var preferred: Level = Level(rawValue: Preference.integer(for: .logLevel).clamped(to: 0...3))!
```

Comment:
I think it's more logical for the clamping to occur in `Level`'s constructor, so we don't have to depend on `Level`'s clients to behave correctly.

Reply:
Use Swift 4.2, which has [SE-0194](https://github.com/apple/swift-evolution/blob/master/proposals/0194-derived-collection-of-enum-cases.md).

Reply:
Then let's put it on `mojave` branch.